atlas-itests-parent/atlas-itests-core/src/test/java/io/atlasmap/core/issue/CamelAtlasmap14Test.java

```diff
@@ -0,0 +1,50 @@
+package io.atlasmap.core.issue;
+
+import java.net.URL;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.atlasmap.api.AtlasContext;
+import io.atlasmap.api.AtlasSession;
+import io.atlasmap.core.AtlasMappingService;
+import io.atlasmap.core.AtlasMappingService.AtlasMappingFormat;
+import io.atlasmap.core.DefaultAtlasContextFactory;
+import io.atlasmap.v2.AtlasMapping;
+import io.syndesis.connector.salesforce.Contact;
+import twitter4j.Status;
+
+public class CamelAtlasmap14Test {
+
+    private AtlasMappingService mappingService;
+    
+    @Before
+    public void before() throws Exception {
+        mappingService = DefaultAtlasContextFactory.getInstance().getMappingService();
+    }
+    
+    @Test
+    public void test() throws Exception {
+        URL url = Thread.currentThread().getContextClassLoader().getResource("mappings/issue-camel-atlasmap-14-mapping.json");
+        AtlasMapping mapping = mappingService.loadMapping(url, AtlasMappingFormat.JSON);
+        AtlasContext context = DefaultAtlasContextFactory.getInstance().createContext(mapping);
+        AtlasSession session = context.createSession();
+        session.setInput(generateTwitterStatus());
+        context.process(session);
+        Object output = session.getOutput();
+        Assert.assertEquals(Contact.class, output.getClass());
+        Contact contact = (Contact)output;
+        Assert.assertEquals("bobvila1982", contact.getTwitterScreenName__c());
+    }
+    
+    protected Status generateTwitterStatus() {
+        MockStatus status = new MockStatus();
```

Comment:
wouldn't it be easier to use a mocking lib like jmockit or mockito to build up a mock object ? Or is this something which doesn't really work because atlasmap uses reflection internally which can't be provided by a dynamic proxy as used by those mock libs ? (guess the later is the case)

Reply:
It may be worth a try - I just copied this mock from another testcase.